tests/test_models/test_mixins.py

```diff
@@ -124,3 +149,90 @@ def test_save_mixin_load_warning(get_version_mock, save_version, load_version, t
     ):
         get_version_mock.return_value = load_version
         _ = DummyNN.load(path)
+
+
+@pytest.mark.parametrize(
+    "mixin_constructor, call_params",
+    [
+        (PredictionIntervalContextIgnorantModelMixin, {}),
+        (NonPredictionIntervalContextIgnorantModelMixin, {}),
+        (PredictionIntervalContextRequiredModelMixin, {"prediction_size": 10}),
+        (NonPredictionIntervalContextRequiredModelMixin, {"prediction_size": 10}),
+    ],
+)
+@pytest.mark.parametrize("return_components", (True, False))
+def test_model_mixins_calls_add_target_components_in_forecast(mixin_constructor, return_components, call_params):
+    with patch.multiple(mixin_constructor, __abstractmethods__=set()):
+        ts = Mock()
+        forecast_ts = Mock(spec=TSDataset)
+        mixin = mixin_constructor()
+        mixin._forecast = Mock(return_value=forecast_ts)
+        mixin._add_target_components = Mock()
+
+        _ = mixin.forecast(ts=ts, return_components=return_components, **call_params)
+
+        mixin._add_target_components.assert_called_with(
+            ts=ts,
+            predictions=forecast_ts,
+            components_prediction_method=mixin._forecast_components,
+            return_components=return_components,
+        )
+
+
+@pytest.mark.parametrize(
+    "mixin_constructor, call_params",
+    [
+        (PredictionIntervalContextIgnorantModelMixin, {}),
+        (NonPredictionIntervalContextIgnorantModelMixin, {}),
+        (PredictionIntervalContextRequiredModelMixin, {"prediction_size": 10}),
+        (NonPredictionIntervalContextRequiredModelMixin, {"prediction_size": 10}),
+    ],
+)
+@pytest.mark.parametrize("return_components", (True, False))
+def test_model_mixins_calls_add_target_components_in_predict(mixin_constructor, return_components, call_params):
+    with patch.multiple(mixin_constructor, __abstractmethods__=set()):
+        ts = Mock()
+        predict_ts = Mock(spec=TSDataset)
+        mixin = mixin_constructor()
+        mixin._predict = Mock(return_value=predict_ts)
+        mixin._add_target_components = Mock()
+        _ = mixin.predict(ts=ts, return_components=return_components, **call_params)
+
+        mixin._add_target_components.assert_called_with(
+            ts=ts,
+            predictions=predict_ts,
+            components_prediction_method=mixin._predict_components,
+            return_components=return_components,
+        )
+
+
+@pytest.mark.parametrize("mixin_constructor", [PerSegmentModelMixin])
+def test_make_prediction_segment_with_components(mixin_constructor, target_components_df, ts_without_target_components):
```

Comment:
We haven't changed `_make_predictions_segment` in this task, do we really need this test? 

Should we add other tests on this method as well (no components, handling `prediction_size`, etc)?

Reply:
Here we check how it works with target components